samples/excel/89-preview-apis/chart-trendlines.yaml

```diff
@@ -0,0 +1,203 @@
+name: Chart trendlines
+description: 'Add, get, set and format trendlines to a chart.'
+host: EXCEL
+api_set:
+    ExcelAPI: 1.7
+script:
+    content: |+
+        $("#setup").click(() => tryCatch(setup));
+        $("#add-trendline-polynomial").click(() => tryCatch(addTrendlinePolynomial));
+        $("#get-trendline").click(() => tryCatch(getTrendlineType));
+        $("#get-trendline-color").click(() => tryCatch(getTrendlineColor));
+        $("#set-trendline-color").click(() => tryCatch(setTrendlineColor));
+
+        async function addTrendlinePolynomial() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+
+                let seriesCollection = sheet.charts.getItemAt(0).series;
+
+                 // Add a trendline of type polynomial.  
+                seriesCollection.getItemAt(0).trendlines.add("Polynomial");
+                let trendline = seriesCollection.getItemAt(0).trendlines.getItem(0);
+                trendline.polynomialOrder = 5;        
+                  
+                await context.sync();        
+            });
+        }
+
+        async function getTrendlineType() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+
+                let seriesCollection = sheet.charts.getItemAt(0).series;
+
+                // Get the type of chart trendline.  
+                // In this sample, it's polynomial.
+                let trendline = seriesCollection.getItemAt(0).trendlines.getItem(0);
+                trendline.load("type");
+                
+                await context.sync();   
+
+                console.log("The trendline type is:" + trendline.type);  
+            });
+        }
+
+        async function getTrendlineColor() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+
+                let seriesCollection = sheet.charts.getItemAt(0).series; 
+
+                // Get the color used to draw the chart trendline.  
+                // In this example, it's #4472c4 which is blue.
+                let trendline = seriesCollection.getItemAt(0).trendlines.getItem(0);
+                let line = trendline.format.line;
+                line.load("color");
+
+                await context.sync();
+                console.log("The trendline color is:" + line.color);
+            });
+        }
+
+        async function setTrendlineColor() {
+            await Excel.run(async (context) => {
+                const sheet = context.workbook.worksheets.getItem("Sample");
+
+                let seriesCollection = sheet.charts.getItemAt(0).series; 
+                let trendline = seriesCollection.getItemAt(0).trendlines.getItem(0);
+
+                // Format the trendline to a solid line,
+                // and set the color used to draw the chart trendline.
```

Comment:
Change comment to:

// Set format of the trendline to a solid, red line.

Reply:
ok